Error ready_event.cc

**Environment:**
1. Framework: (TensorFlow, Keras, PyTorch, MXNet)
2. Framework version:
3. Horovod version:
4. MPI version:
5. CUDA version:
6. NCCL version:
7. Python version:
8. Spark / PySpark version:
9. OS and version:
10. GCC version:
11. CMake version:

**Checklist:**
1. Did you search issues to find if somebody asked this question before?

2. If your question is about hang, did you read [this doc](https://github.com/horovod/horovod/blob/master/docs/running.rst)?
3. If your question is about docker, did you read [this doc](https://github.com/horovod/horovod/blob/master/docs/docker.rst)?
4. Did you check if you question is answered in the [troubleshooting guide](https://github.com/horovod/horovod/blob/master/docs/troubleshooting.rst)?

**Bug report:**
Please describe erroneous behavior you're observing and steps to reproduce it.

Build error in 0.20.1

01-install-python.sh    horovod>=v0.20.1          /tmp/pip-install-y4hzs7wm/horovod/horovod/torch/ready_event.cc: In function ‘std::shared_ptr<horovod::common::ReadyEvent> horovod::torch::RecordR
eadyEvent(int)’:                                                                                                                                                                                   
01-install-python.sh    horovod>=v0.20.1          /tmp/pip-install-y4hzs7wm/horovod/horovod/torch/ready_event.cc:109:16: error: ‘logic_error’ is not a member of ‘std’                             
01-install-python.sh    horovod>=v0.20.1            109 |     throw std::logic_error("Internal error. Requested ReadyEvent "                                                                       
01-install-python.sh    horovod>=v0.20.1                |                ^~~~~~~~~~~                                                                                                               
01-install-python.sh    horovod>=v0.20.1          /tmp/pip-install-y4hzs7wm/horovod/horovod/torch/ready_event.cc:113:1: warning: control reaches end of non-void function [-Wreturn-type]          
01-install-python.sh    horovod>=v0.20.1            113 | }                                                                                                                                        
01-install-python.sh    horovod>=v0.20.1                | ^

--- a/horovod/torch/cuda_util.cc
+++ b/horovod/torch/cuda_util.cc
@@ -16,6 +16,8 @@
 #if HAVE_GPU
 #include "cuda_runtime.h"
 #include <THC/THC.h>
+#else
+#include <stdexcept>
 #endif
 
 #include "../common/common.h"

--- a/horovod/torch/ready_event.cc
+++ b/horovod/torch/ready_event.cc
@@ -24,6 +24,8 @@
 #include <mutex>
 #include <queue>
 #include <unordered_map>
+#else
+#include <stdexcept>
 #endif
 
 #include "ready_event.h"

